tests/functionality_tests/test_activation_shapes.py

```diff
@@ -18,13 +17,26 @@ def main():
         print("flip rows, cols for traversal")
 
     activation_type = input("Activation type: ")
-    is_vertical = "True" == input("Vertical traversal: ")
-    if is_vertical:
+    direction = input("Traversal Direction: ")
+    if direction.isnumeric():
+        numberToDirection = {"0": "RIGHT", "1": "UP", "2": "LEFT", "3": "DOWN"}
```

Comment:
im ngl, i dont know if this is necessary to add; i dont know who would associate 0 to right, 1 to up, etc

Reply:
It doesn't hurt to keep it? Also credits to Ahan! I test_activation_shapes.py from his PR and it's great.

Reply:
ig not, i dont know if it'll crowd the code space